### DISCUSSION TITLE
array_filter(): Argument #1 ($array) must be of type array, int given

### DISCUSSION THREAD

**Discussion by mohjacksi** [spatie/laravel-translatable#339] [Q&A] [answered]
Hello,
I'm getting this error when I use more thank one filed in $translatable array!
it work perfect when I just use one element for translation like:     public $translatable = ['name'];
But when I use more than one:
    public $translatable = [
        'name',
        'notes',
        'about',
        'location',
    ];

It give me this error when I want to access other fields except the first one (works perfectly)
array_filter(): Argument #1 ($array) must be of type array, int given

**Comment by anibalealvarezs** [comment]
Hello. Could you please share the whole error message? Include de line and document where it's happening.

**Reply by mohjacksi** [reply]
Sure, https://flareapp.io/share/dmkvw4w5#share

**Reply by mohjacksi** [reply]
BTW, I tried to delete (name) and everything else from translatable and just leave (notes), it shows me the same error
and my database already has data inside it

**Comment by anibalealvarezs** [comment, marked as answer]
Your problem is that [here](https://github.com/spatie/laravel-translatable/blob/main/src/HasTranslations.php#L96) the ```getAttributes()``` method is returning an integer.

It's not the problem with the amount of fields you're translating, but with the value you're getting from one of the fields in the array.

Since the problem happens when getting the translation, and since ```name``` works, I assume the problem is one of the other 3 fields, which is not returning an string, or the translation doesn't exist.

 - Start by removing one field at a time from the array to find which of them is causing the issue.
 - Once you had found it, check the column format in the database an make sure you get an string and confirm that you're not mutating the field by other means in the model.
 - If it still fails, remove it from the ```translatable``` array and ```var_dump()``` the model, so you can see exactly what the value and field format are.

Good coding!

**Reply by mohjacksi** [reply]
Many thanks bro, I'll check it right now!

**Reply by mohjacksi** [reply]
The problem was the fields had 123,12451,1241 (numeric data) but the type was string in database! when I edited the fields data to some dfasfa,fafa,fdd it worked for me!
I'm good now, but I want to know why this happens ?

**Reply by anibalealvarezs** [reply]
Well, check model's insertions/updates along the code. That's a bit harder to find.
There are plenty of ways for inserting records in your tables.

Make sure those fields are no meant to store relations (which would make them store just the ID of the related model).

Also make sure that, in your model, you don't have a mutator for setting the value incorrectly.

**Comment by x0xa** [comment]
Hello
Have a same error with string numbers data, for example:

`dump(json_decode("boom", true), json_decode("'1111'", true), json_decode(1111, true), json_decode("1111", true));`
`null`
`null`
`1111`
`1111`

in two last cases we got error

`array_filter(): Argument #1 ($array) must be of type array, int given`

cause json_decode returns number, not null

getAttributes() method is returning an string with number (last case in my example)
model have not any mutators or casts

[db screen](https://i.imgur.com/Hv1TBNC.png)

1 works fine
2 got error

Thank you!
